test/fixtures.py

```diff
@@ -62,20 +63,37 @@ def pylsp_w_workspace_folders(tmpdir):
     return (ls, workspace_folders)
 
 
+@pytest.fixture()
+def dispatcher():
+    return {}
+
+
+@pytest.fixture()
+def consumer():
+    return MagicMock()
+
+
+@pytest.fixture()
+def endpoint(dispatcher, consumer):
+    return Endpoint(dispatcher, consumer, id_generator=lambda: "id")
```

Comment:
```suggestion
@pytest.fixture()
def endpoint(dispatcher, consumer):
    return Endpoint({}, MagicMock(), id_generator=lambda: "id")
```

I don't see the first two fixtures as necessary because they're not used by anything else.